### ISSUE TITLE
Error on runtime and on newer builds

### ISSUE THREAD

**Issue by RGFTheCoder** [not-an-aardvark/lucky-commit#10]
I get this error (repeated 15 times) when running the program (2.0.1) or building it on the latest version:
```
<kernel>:107:17: error: subscripted access is not allowed for OpenCL vectors
    uint w0_t = w[0];
```

**Comment by not-an-aardvark** [comment]
Thanks for reporting. I guess this is a compatibility issue between different versions of the OpenCL compiler.

Does it work if you build from the [`no-subscript-vector-access` branch](https://github.com/not-an-aardvark/lucky-commit/tree/no-subscript-vector-access) (currently at commit 00000009bb9341332482cab16e1d868f07fa776c)?

**Comment by birkenfeld** [comment]
I had the same build errors about subscripted access. With `no-subscript-vector-access` these are gone but I get:
```
error: failed to compile `lucky_commit v2.0.3 (https://github.com/not-an-aardvark/lucky-commit/?branch=no-subscript-vector-access#00000009)`, intermediate artifacts can be found at `/tmp/cargo-install6FPJLc`

Caused by:
  failed to run custom build command for `lucky_commit v2.0.3 (/home/gbr/.cargo/git/checkouts/lucky-commit-4fb4b0489090f150/00000009bb)`

Caused by:
  process didn't exit successfully: `/tmp/cargo-install6FPJLc/release/build/lucky_commit-cc2f1e47141252e2/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-changed=build.rs

  --- stderr
  thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: 

  ###################### OPENCL PROGRAM BUILD DEBUG OUTPUT ######################

  <kernel>:18:12: error: __kernel function cannot have argument whose type is, or contains, type size_t
      size_t num_dynamic_blocks,
             ^
  
  ###############################################################################

  ', build.rs:23:10
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by not-an-aardvark** [comment]
Thanks. How about the updated version of the `no-subscript-vector-access` branch (commit 00000006126d571472b06e794bf550b93522e95b)?

I'm not sure of the best way to test this with other OpenCL compilers, so I appreciate your help. Hopefully there aren't too many more errors to resolve.

**Comment by birkenfeld** [comment]
That one works perfectly!

All those commit references are hilarious to look at btw :)

**Comment by not-an-aardvark** [comment]
Great! Merged those two commits onto `main` as 0000000786b8032dfe8b57717b215a51789386cb and 0000000088d59d4d1273d70686e61449cd130f9b.
